### DISCUSSION TITLE
Are images within svgs possible?

### DISCUSSION THREAD

**Discussion by Bobakanoosh** [ssssota/svg2png-wasm#60] [Q&A] [answered]
I need to put an image in my svg, and have tried the following methods:
```html
  // imgSrc here is a base64 representation of an image
  <image xlink:href="data:image/jpeg;base64,${imgSrc}" x="200" y="20.0" width="100" />
```
and
```html
    <image href="https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png" x="200" y="20.0" width="100" height=""/>
```

With no success. If I paste the svg before it's converted into a codepen, everything works fine. The other elements I have in the svg (e.g. text) still render fine.

Is this a limitation of the library?

**Comment by ssssota** [comment, marked as answer]
Thank you for your question!

As you say, image rendering is not currently supported.

> Is this a limitation of the library?

I think the answer is "no".
However, there is no conclusive proof.

svg2png-wasm is made with `resvg`(Rust crate). And `resvg` can render images.
So I'll try to support image rendering!

**Reply by Bobakanoosh** [reply]
Awesome, being able to use this for images would be amazing!

I want to use it to generate open graph images dynamically for site meta tags. I've got everything working except I need to put a user's avatar on there as well.

**Reply by dcdunkan** [reply]
Thank you, supporting images would be great and useful!
